File: 1545-find-kth-bit-in-nth-binary-string/1545-find-kth-bit-in-nth-binary-string.cpp
class Solution {
public:
    string invert(string s){
        for(int i=0; i<s.length(); i++){
            if(s[i] == '0'){
                s[i] = '1';
            }
            else{
                s[i] = '0';
            }
        }
        return s;
    }
    char findKthBit(int n, int k) {
        // char ans = '';
        string s = "0";
        for(int i=0; i<n-1; i++){
            string temp = s;
            string in = invert(temp);
            reverse(in.begin(), in.end());
            s = temp+"1"+in;
        }
        return s[k-1];
    }
};

File: 1545-find-kth-bit-in-nth-binary-string/1545-find-kth-bit-in-nth-binary-string.js
/**
 * @param {number} n
 * @param {number} k
 * @return {character}
 */
var getBit = function(i, s, target){
    if(i > target) return s;
    var temp = "";
    for(var j=0; j<s.length; j++){
        temp += s[j] === '0' ? '1' : '0';
    }
    s += ('1');
    s += temp.split('').reverse().join('');
    return getBit(i+1, s, target);
}

var findKthBit = function(n, k) {
    var s = "0";
    s = getBit(2, s, n);
    return s[k-1];
};